### ISSUE TITLE
Crash/incorrect handling of materials

### ISSUE THREAD

**Issue by Paril** [jdolan/quetoo#472]
![ep3ftanr](https://cloud.githubusercontent.com/assets/643118/25851140/94719924-3492-11e7-8946-7aa16d64a34e.png)

```
R_RegisterDependency: textures/edge/rockies_mat -> textures/edge/mossyrockies_mat
R_LoadMaterials: Parsed material edge/rockies with 1 stages
Cm_ResolveMaterialAsset: Resolved textures/edge/mossyrockies.tga for edge/mossyrockies
R_RegisterMedia: Replacing textures/edge/mossyrockies_mat
R_FreeMedia: Freeing textures/edge/mossyrockies_mat
R_RegisterDependency: textures/edge/mossyrockies_mat -> textures/edge/mossyrockies
R_RegisterDependency: textures/edge/mossyrockies_mat -> textures/edge/mossyrockies_nm
R_LoadMaterials: Parsed material edge/mossyrockies with 0 stages
Cm_ResolveMaterialAsset: Resolved textures/edge/metl12_1.tga for edge/metl12_1
```
Crash on edge with very high repro rate. Related to freemedia on mossyrockies - rockies needs this texture.

**Comment by jdolan** [comment]
For context, in case anyone feels like working on this before I can get to it: this is a regression from the recent materials refactor.

**Comment by Paril** [comment]
Also, footstep sounds and default colors for skins are broken.

**Comment by kaadmy** [comment]
This sounds like material files aren't being loaded correctly.

**Comment by Paril** [comment]
Fixed footstep sounds in 455eb7d (.mat file it looked for forgot to strip extension, so it was looking for edge.bsp.mat).

Only thing left now is player model materials don't load. Things of note:
- R_LoadMd3Model has it explicitly set not to load .mat files from players/ - this will prevent materials inside those files from being registered at all. I don't think this code should be there. The odd thing is, if you change this so it *can* load from player models, you then get an ERROR_DROP about a null diffuse name. Reason is that R_LoadMeshMaterials only really works with non-player models (or possibly only if the skin file actually exists? dragoon, for instance, has no lower.mat, so it fails loading that, then tries to create a fake material for players/dragoon/skin, but "diffuse" is never filled in so it crashes in R_ResolveMaterial trying to resolve an empty image name. It seems like the bigger issue is just that R_LoadMeshMaterials will fail if the texture doesn't exist/material doesn't exist for the given model. Player models don't *need* to try to load /skin in worst case scenario since it's the only type of model that has special material handling, but even if it does, it should never crash trying to find a texture that doesn't exist. Indeed, an assert in R_LoadMeshMaterials suggests that at least one material should always be loaded for a model, which makes sense, but it can't do that if cm->diffuse is empty)
- Cg_LoadClientSkin, which pops in the skin r_material_t references, uses LoadMaterial directly (with player asset context) to search through loaded materials to find the ones to use. If the material doesn't exist, then R_LoadMaterial makes one that just searches for the base stuff, which is why it works currently for player models. The tintmap is implicit (unless explicit), so having the tintmap work does not mean materials are working: the helmets for guard and qforcer, for instance, are opaque now because it isn't loading the material file.
